### PR TITLE
[13.0][FIX] delivery_tnt_oca: Encode data to utf-8 to prevent error in some cases (Šempas for example)

### PR DESCRIPTION
**Pull request by victoralmau**
Encode data to utf-8 to prevent error in some cases (Šempas for example)

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT37181

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Documentation says that requests are handled in that encoding? Isn't this something that can be configure at request level instead of directly encoding? And shouldn't an intermediate method should handle this for doing the conversion for all requests?

**Comment by victoralmau** [comment]
In one use case, the API returned the following error: "_UnicodeEncodeError: 'latin-1' codec can't encode character '\u0160' in position 1873: Body ('Š') is not valid Latin-1. Use body.encode('utf-8') if you want to send it encoded in UTF-8._"

Ok, if you prefer we do it at the request level directly.

**Comment by pedrobaeza** [comment]
Waiting for the other change.

**Comment by victoralmau** [comment]
> Documentation says that requests are handled in that encoding? Isn't this something that can be configure at request level instead of directly encoding? And shouldn't an intermediate method should handle this for doing the conversion for all requests?

The applied change utf8-encodes the data for all requests just before sending it. Do you think it should be done differently?

File: delivery_tnt_oca/models/tnt_request.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ def _send_api_request(
             headers = {"Content-Type": content_type}
```
Comment (pedrobaeza) [inline delivery_tnt_oca/models/tnt_request.py]: Looking over Internet, it seems you have to put it this way:

```suggestion
            headers = {"Content-Type": content_type; "charset": "UTF-8"}
```

File: delivery_tnt_oca/models/tnt_request.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def _send_api_request(
         if data is None:
             data = {}
         try:
-            headers = {"Content-Type": content_type}
+            headers = {"Content-Type": content_type, "charset": "UTF-8"}
             if auth:
                 headers["Authorization"] = "Basic {}".format(self.authorization)
             res = requests.post(url=url, data=data, headers=headers, timeout=60)
```
Comment (pedrobaeza) [inline delivery_tnt_oca/models/tnt_request.py]: But you still require to encode `data`:

```suggestion
            res = requests.post(url=url, data=data.encode("utf-8"), headers=headers, timeout=60)
```

**Review by pedrobaeza — APPROVED** [review]

**Comment by victoralmau** [comment]
Ping @sergio-teruel

**Review by cesar-tecnativa — APPROVED** [review]

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**